### PR TITLE
staking-async: handle uninitialized state in try-state checks

### PR DESCRIPTION
**Pull request by sigurpol**
- Add early return in do_try_state when pallet is uninitialized
- Add test for empty state validation

Followup of #9721 .
Once backported to `2507` and crate is published, should unlock https://github.com/polkadot-fellows/runtimes/pull/904

### CODE REVIEW

File: substrate/frame/staking-async/src/pallet/impls.rs [hunk 1]
```diff
@@ -1942,7 +1949,13 @@ impl<T: Config> Pallet<T> {
 		// Sanity check for the paged exposure of the active era.
 		let mut exposures: BTreeMap<T::AccountId, PagedExposureMetadata<BalanceOf<T>>> =
 			BTreeMap::new();
-		let era = ActiveEra::<T>::get().unwrap().index;
+		// If the pallet is not initialized, we return immediately from pallet's do_try_state() and
+		// we don't call this method. Otherwise, Eras::do_try_state enforces that both ActiveEra
+		// and CurrentEra are Some. Thus, we can safely unwrap here.
+		let era = ActiveEra::<T>::get()
+			.expect("ActiveEra must be set when checking paged exposures")
```
Comment (bkontur) [inline substrate/frame/staking-async/src/pallet/impls.rs]: maybe nit (I know it is just try-runtime, but :)):
```suggestion
			.ok_or(TryRuntimeError::Other("ActiveEra must be set when checking paged exposures"))?
```

**Review by bkontur — APPROVED** [review]

File: substrate/frame/staking-async/src/pallet/impls.rs (hunk 1 above)
Comment (sigurpol) [reply substrate/frame/staking-async/src/pallet/impls.rs]: I actually had `ok_or()` initially, but then I had a philosophical doubt if for an invariant in try runtime, we should assert instead of returning an error. I thought that asserting was better in this case but I am totally fine to switch to the `ok_or` way. Let me try to see if democracy is still alive these days and summon Kian as well:  @kianenigma  what do you think it's the cleaner approach here?

Comment (kianenigma) [reply substrate/frame/staking-async/src/pallet/impls.rs]: Not a hill I want to die on but: 

1. While I personally prefer `.unwrap()` / `expect()` in such code paths
2. We do have `-> Result<(), ()>` in the `try_state` code path, so I'd respect the overarching return value and do my best to return errors rather than panic.

Comment (sigurpol) [reply substrate/frame/staking-async/src/pallet/impls.rs]: Ubi maior, minor cessat 😄
